### ISSUE TITLE
Support for dumping into non-empty repositories.

### ISSUE THREAD

**Issue by drt24** [wez/svn-to-git#1]
If the svn dump file is a continuation of the history already stored in the git repository specified with --git-repo then svn-to-git does not use the pre-existing information about the files that already exist (probably because this is not implemented). This results in errors such as:
"
PHP Fatal error:  Uncaught exception 'Exception' with message 'build.xml not found in ' in /home/daniel/dev/svn-to-git/util.php:177
Stack trace:
#0 /home/daniel/dev/svn-to-git/util.php(458): SvnFile->resolveForWrite('build.xml')
#1 /home/daniel/dev/svn-to-git/svn-to-git(477): Repo->discover(Object(SvnDumpReader))
#2 {main}

  thrown in /home/daniel/dev/svn-to-git/util.php on line 177
"
when trying to do this.

Now I know this is rather an odd thing to be trying to do but I am trying to splice some history stored in SVN onto the older history stored in git (the person who stored the history in SVN did the work off a tarball rather than off the repository).
svn-to-git has got me the closest to actually being able to do this of all the different importers I have tried.

Thank you for making this program.

**Comment by wez** [comment]
Can you expand on the relative ordering of these bits?
It sounds like you started on git, someone forked a bit and tracked some work in svn, and now you want to reconcile it into your existing git repo.  Is that correct?

**Comment by drt24** [comment]
There is slightly more detail about what I was trying to do here: https://www.srcf.ucam.org/~drt24/blog/2010/06/major-multi-vcs-surgery/
but to summarise:
10 years ago development using a CVS repository halted and no development occurred for 9 years. Last year someone took a source tarball from that and tracked their changes in SVN for a few months before halting work on that project. This summer I started work from a conversion of the CVS repository to Git and did several weeks work on it. Then I found out about the work that this other person had done and tracked in SVN. What I wanted to do was to import their history into a separate git branch beginning at the common part of history that was in the old CVS repository as this might have made merging slightly neater.
So

<pre>
 CVS
 |         \
 SVN    Git
    |    /
    Git
</pre>

Is what I am trying to do.

**Comment by wez** [comment]
I'm not so sure that it is as simple as teaching svn-to-git about what is already in the git repo, or perhaps that is better worded as, it is going to be simple to teach svn-to-git to read existing git history; it wasn't built to work that way.

To guarantee the right results, I think you need to convert the CVS repo to SVN and splice the SVN changes into that repo.  You can then svn-to-git from there, and you should have yourself a git repo that you can merge into your own git repo.

If I were in your shoes, I'd consider looking at Mercurial as a means to this end; its convert extension has some splicing support that might do the heavy lifting you need.  From there, you can either go hg->git or hg->svn and svn-to-git the result.  I haven't tried this, so I can't provide much commentary on which path might be the better one.

Hope that helps!

**Comment by wez** [comment]
ugh, I meant: "it isn't going to be simple".

**Comment by drt24** [comment]
Yeah I did the convert CVS repo to SVN and then import trick but unfortunately Git's method of using hashes of the the repository for commits means that despite actually being identical I get different hashes due to slight differences in metadata etc...

The hg option is interesting: I might try that but I suspect I will end up facing the slight differences due to different conversion paths resulting in different hashes problem again.

I think that if svn-to-git simply knew the current state of the repository then it might be able to append commits properly but then again I didn't write it so I don't know :-)

Thanks anyway.
